lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -2677,6 +2677,19 @@ bool SIInstrInfo::isAlwaysGDS(uint16_t Opcode) const {
          Opcode == AMDGPU::DS_GWS_BARRIER;
 }
 
+bool SIInstrInfo::opcodeEmitsNoInsts(const MachineInstr &MI) const {
```

Comment:
Since this function is called in multiple places from different passes, moved it to SIInstrInfo.